eslzArm/managementGroupTemplates/policyAssignments/ENFORCE-ALZ-DecommissionedPolicyAssignment.json

```diff
@@ -0,0 +1,70 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "topLevelManagementGroupPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Provide the ESLZ company prefix to the intermediate root management group containing the policy definitions."
+            }
+        },
+        "enforcementMode": {
+            "type": "string",
+            "allowedValues": [
+                "Default",
+                "DoNotEnforce"
+            ],
+            "defaultValue": "Default"
+        },
+        "listOfResourceTypesAllowed": {
+            "type": "Array",
+            "defaultValue": [
+                "microsoft.consumption/tags",
+                "microsoft.authorization/roleassignments",
+                "microsoft.authorization/roledefinitions",
+                "microsoft.authorization/policyassignments",
+                "microsoft.authorization/locks",
+                "microsoft.authorization/policydefinitions",
+                "microsoft.authorization/policysetdefinitions",
+                "microsoft.resources/tags",
+                "microsoft.authorization/roleeligibilityschedules",
+                "microsoft.authorization/roleeligibilityscheduleinstances",
+                "microsoft.authorization/roleassignmentschedules",
+                "microsoft.authorization/roleassignmentscheduleinstances"
+            ]
+        }
+    },
+    "variables": {
+        "policyDefinitions": {
+            "enforceAlzDecommissioned": "[concat('/providers/Microsoft.Management/managementGroups/', parameters('topLevelManagementGroupPrefix'), '/providers/Microsoft.Authorization/policySetDefinitions/Enforce-ALZ-Decomm')]"
+        },
+        "policyAssignmentNames": {
+            "alzDecommission": "Enforce-ALZ-Decomm",
+            "description": "This initiative will help enforce and govern subscriptions that are placed within the decommissioned Management Group as part of your Subscription decommissioning process. See https://aka.ms/alz/policies for more information.",
+            "displayName": "Enforce ALZ Decommissioned Guardrails"
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Authorization/policyAssignments",
+            "apiVersion": "2019-09-01",
+            "name": "[variables('policyAssignmentNames').alzDecommission]",
+            "location": "[deployment().location]",
+            "identity": {
+                "type": "SystemAssigned"
+            },
```

Comment:
This is needed on this one as you have DINE/Modify effects in the child definitions.

However, there is no RBAC assignment of the Managed Identity that get created for the policy assignment, so the remediation tasks will fail.

Example of what you need to add to this ARM template for the assignment to is here: https://github.com/Azure/Enterprise-Scale/blob/05294194ed22510b0d3635766eeb3d6f0f127f3d/eslzArm/managementGroupTemplates/policyAssignments/DINE-ResourceDiagnosticsPolicyAssignment.json#L61-L73

Reply:
Well spotted.

Reply:
Good catch. Resolved to support remediation tasks.